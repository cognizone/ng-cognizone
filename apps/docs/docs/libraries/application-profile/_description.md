This library provides models and utility functions for the ApplicationProfile type and related logic.
