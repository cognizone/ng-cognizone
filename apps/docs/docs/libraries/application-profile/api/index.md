---
id: "index"
title: "@cognizone/application-profile"
slug: "/libraries/application-profile/api/"
sidebar_label: "Readme"
sidebar_position: 0
custom_edit_url: null
---

# application-profile

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test application-profile` to execute the unit tests.
