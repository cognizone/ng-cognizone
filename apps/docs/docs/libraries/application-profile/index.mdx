---
id: 'index'
title: '@cognizone/application-profile'
slug: '/libraries/application-profile/'
sidebar_label: 'Readme'
sidebar_position: 0
---

import Description from './_description.md';

<Description />
