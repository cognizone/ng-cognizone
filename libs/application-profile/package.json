{
  "name": "@cognizone/application-profile",
  "version": "2.4.3",
  "peerDependencies": {
    "@cognizone/model-utils": "^1.28.0"
  },
  "dependencies": {
    "tslib": "^2.0.0"
  },
  "license": "Apache-2.0",
  "homepage": "https://github.com/cognizone/ng-cognizone",
  "repository": {
    "type": "git",
    "url": "https://github.com/cognizone/ng-cognizone.git",
    "directory": "libs/application-profile"
  },
  "contributors": [
    {
      "name": "William Chelman",
      "email": "william.chelman@cogni.zone"
    },
    {
      "name": "Farah El Agha",
      "email": "farah.agha111@gmail.com"
    }
  ],
  "private": false,
  "publishConfig": {
    "access": "public"
  }
}