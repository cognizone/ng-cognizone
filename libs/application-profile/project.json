{
  "name": "application-profile",
  "sourceRoot": "libs/application-profile/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/application-profile",
        "main": "libs/application-profile/src/index.ts",
        "tsConfig": "libs/application-profile/tsconfig.lib.json",
        "assets": ["libs/application-profile/*.md"]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/application-profile/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/application-profile"],
      "options": {
        "jestConfig": "libs/application-profile/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
