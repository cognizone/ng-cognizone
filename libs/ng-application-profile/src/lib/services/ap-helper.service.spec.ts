/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable @typescript-eslint/no-require-imports */
import { ApplicationProfile, EMPTY_APPLICATION_PROFILE } from '@cognizone/application-profile';
import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { ApHelper } from './ap-helper.service';
import { ApService } from './ap.service';

const directiveAp: ApplicationProfile = require('../../test/directive-ap.json');

describe('ApHelper', () => {
  let spectator: SpectatorService<ApHelper>;
  const createService = createServiceFactory({ service: ApHelper, mocks: [ApService] });

  beforeEach(() => (spectator = createService()));

  test('should be injectable', () => {
    expect(spectator.service).not.toBeUndefined();
  });

  test('should get Event type', () => {
    const profile = spectator.service.getTypeProfile(directiveAp, ['EUDossier', 'Event']);
    expect(profile).toMatchSnapshot();
  });

  test('should get EUDossier type', () => {
    const profile = spectator.service.getTypeProfile(directiveAp, 'EUDossier');
    expect(profile).toMatchSnapshot();
  });

  test('should throw if type not found', () => {
    // const profile = spectator.service.getTypeProfile(EMPTY_APPLICATION_PROFILE, 'EUDossier');
    expect(() => spectator.service.getTypeProfile(EMPTY_APPLICATION_PROFILE, 'EUDossier')).toThrow();
  });

  test('should get concrete type', () => {
    expect(spectator.service.getConcreteType(directiveAp, ['Event', 'Transposition'])).toBe('Transposition'); // exact match
    expect(spectator.service.getConcreteType(directiveAp, 'Transposition')).toBe('Transposition'); // just the concrete one
  });
});
