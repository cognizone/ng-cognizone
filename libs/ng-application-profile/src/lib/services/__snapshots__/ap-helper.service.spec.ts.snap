// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApHelper should get EUDossier type 1`] = `
Object {
  "attributes": Object {
    "contentsEvent": Object {
      "attributeId": "contentsEvent",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUEventLegal",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierContentsEvent",
    },
    "euDossierProducesResourceLegal": Object {
      "attributeId": "euDossierProducesResourceLegal",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EULegalResource",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierProducesResourceLegal",
    },
    "euDossierResponsibilityOfAgent": Object {
      "attributeId": "euDossierResponsibilityOfAgent",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EULegalInstitution",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierResponsibilityOfAgent",
    },
    "eventRiskLevel": Object {
      "attributeId": "eventRiskLevel",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2000/01/rdf-schema#Resource",
          },
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventRiskLevel",
    },
    "hasDocument": Object {
      "attributeId": "hasDocument",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Document",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#hasDocument",
    },
    "hasEventRemark": Object {
      "attributeId": "hasEventRemark",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Remark",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventHasEventRemark",
    },
    "identifier": Object {
      "attributeId": "identifier",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2001/XMLSchema#string",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierIdentifier",
    },
    "initiatedByActPreparatory": Object {
      "attributeId": "initiatedByActPreparatory",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUPreparatoryAct",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierInitiatedByActPreparatory",
    },
    "initiatesTransposition": Object {
      "attributeId": "initiatesTransposition",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Transposition",
          },
        },
        Object {
          "name": "minCardinality",
          "value": 1,
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierInitiatesTransposition",
    },
    "isAboutConceptTypeDossier": Object {
      "attributeId": "isAboutConceptTypeDossier",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUDossierType",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierIsAboutConceptTypeDossier",
    },
    "relatedToToDo": Object {
      "attributeId": "relatedToToDo",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2000/01/rdf-schema#Resource",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventRelatedToToDo",
    },
    "title": Object {
      "attributeId": "title",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/1999/02/22-rdf-syntax-ns#langString",
          },
        },
        Object {
          "name": "minCardinality",
          "value": 1,
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierTitle",
    },
  },
  "classIds": Array [
    "EUDossier",
  ],
  "rules": Array [
    Object {
      "name": "rdfTypes",
      "value": Array [
        "http://data.legilux.public.lu/resource/ontology/jolux#Event",
        "http://data.legilux.public.lu/resource/ontology/jolux#EUDossier",
      ],
    },
  ],
}
`;

exports[`ApHelper should get Event type 1`] = `
Object {
  "attributes": Object {
    "contentsEvent": Object {
      "attributeId": "contentsEvent",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUEventLegal",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierContentsEvent",
    },
    "euDossierProducesResourceLegal": Object {
      "attributeId": "euDossierProducesResourceLegal",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EULegalResource",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierProducesResourceLegal",
    },
    "euDossierResponsibilityOfAgent": Object {
      "attributeId": "euDossierResponsibilityOfAgent",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EULegalInstitution",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierResponsibilityOfAgent",
    },
    "eventRiskLevel": Object {
      "attributeId": "eventRiskLevel",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2000/01/rdf-schema#Resource",
          },
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventRiskLevel",
    },
    "hasDocument": Object {
      "attributeId": "hasDocument",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Document",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#hasDocument",
    },
    "hasEventRemark": Object {
      "attributeId": "hasEventRemark",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Remark",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventHasEventRemark",
    },
    "identifier": Object {
      "attributeId": "identifier",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2001/XMLSchema#string",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierIdentifier",
    },
    "initiatedByActPreparatory": Object {
      "attributeId": "initiatedByActPreparatory",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUPreparatoryAct",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierInitiatedByActPreparatory",
    },
    "initiatesTransposition": Object {
      "attributeId": "initiatesTransposition",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "Transposition",
          },
        },
        Object {
          "name": "minCardinality",
          "value": 1,
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierInitiatesTransposition",
    },
    "isAboutConceptTypeDossier": Object {
      "attributeId": "isAboutConceptTypeDossier",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "classId",
            "value": "EUDossierType",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierIsAboutConceptTypeDossier",
    },
    "relatedToToDo": Object {
      "attributeId": "relatedToToDo",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/2000/01/rdf-schema#Resource",
          },
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#eventRelatedToToDo",
    },
    "title": Object {
      "attributeId": "title",
      "rules": Array [
        Object {
          "name": "range",
          "value": Object {
            "name": "datatype",
            "value": "http://www.w3.org/1999/02/22-rdf-syntax-ns#langString",
          },
        },
        Object {
          "name": "minCardinality",
          "value": 1,
        },
        Object {
          "name": "maxCardinality",
          "value": 1,
        },
      ],
      "uri": "http://data.legilux.public.lu/resource/ontology/jolux#euDossierTitle",
    },
  },
  "classIds": Array [
    "EUDossier",
  ],
  "rules": Array [
    Object {
      "name": "rdfTypes",
      "value": Array [
        "http://data.legilux.public.lu/resource/ontology/jolux#Event",
        "http://data.legilux.public.lu/resource/ontology/jolux#EUDossier",
      ],
    },
  ],
}
`;
